exchanges/stream/websocket.go

```diff
@@ -210,14 +218,14 @@ func (w *Websocket) Connect() error {
 		w.connectionMonitor()
 	}
 
-	// Resubscribe after re-connection
-	if len(w.subscriptions) != 0 {
-		err = w.Subscriber(w.subscriptions)
-		if err != nil {
-			return fmt.Errorf("%v %w: %v", w.exchangeName, ErrSubscriptionFailure, err)
-		}
+	subs, err := w.GenerateSubs() // regenerate state on new connection
```

Comment:
More of a future warning, but if ever the exchange level `SubscribeToWebsocketChannels` function gets utilised, then this change will not pick up any additional subscriptions/unsubscriptions that have occurred when reconnecting.

Reply:
Would you like me to include fixes for that issue in this PR?

Reply:
Nah, I'll just be lying in wait, ready to nit whoever uses `SubscribeToWebsocketChannels` without updating this section